src/runtime/contrib/dnnl/dnnl_json_runtime.cc

```diff
@@ -150,33 +171,49 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
+  void Conv2d(const size_t& nid, const bool has_relu = false, const bool has_bias = false, const bool has_sum = false) {
     auto node = nodes_[nid];
-
-    // Setup attributes.
     auto data_entry = node.GetInputs()[0];
     auto weight_entry = node.GetInputs()[1];
-    dnnl::memory::dims input_shape = nodes_[data_entry.id_].GetOpShape()[data_entry.index_];
+    dnnl::memory::dims input_shape = nodes_[data_entry.id_].GetOpShape()[data_entry.index_]; 
     dnnl::memory::dims weight_shape = nodes_[weight_entry.id_].GetOpShape()[weight_entry.index_];
     std::vector<std::string> str_strides = node.GetAttr<std::vector<std::string>>("strides");
     std::vector<std::string> str_padding = node.GetAttr<std::vector<std::string>>("padding");
     dnnl::memory::dim groups = std::stoi(node.GetAttr<std::vector<std::string>>("groups")[0]);
+    auto src_df = layout_dict[node.GetAttr<std::vector<std::string>>("data_layout")[0]];
+    auto weight_df = layout_dict[node.GetAttr<std::vector<std::string>>("kernel_layout")[0]];
+    auto dst_df = layout_dict[node.GetAttr<std::vector<std::string>>("out_layout")[0]];
+    std::vector<std::string> outC = node.GetAttr<std::vector<std::string>>("channels");
 
     dnnl::memory::dim N = input_shape[0],       // batch size
-        IC = input_shape[1],                    // input channels
-        IH = input_shape[2],                    // input height
-        IW = input_shape[3],                    // input width
-        OC = weight_shape[0],                   // output channels
-        KH = weight_shape[2],                   // weight height
-        KW = weight_shape[3],                   // weight width
+        IH = input_shape[1],                    // input height
+        IW = input_shape[2],                    // input width
+        IC = input_shape[3],                    // input channels
+        OC = std::stoi(outC[0]),                // output channels
+        KH = weight_shape[1],                   // weight height
+        KW = weight_shape[2],                   // weight width
+        PH_L = std::stoi(str_padding[0]),       // height padding: left
+        PH_R = std::stoi(str_padding[2]),       // height padding: right
         PW_L = std::stoi(str_padding[1]),       // width padding: left
         PW_R = std::stoi(str_padding[3]),       // width padding: right
-        PH_L = std::stoi(str_padding[0]),       // height padding: top
-        PH_R = std::stoi(str_padding[2]),       // height padding: bottom
         SH = std::stoi(str_strides[0]),         // height-wise stride
-        SW = std::stoi(str_strides[1]),         // weight-wise stride
-        OH = (IH - KH + PH_L + PH_R) / SH + 1,  // output height
-        OW = (IW - KW + PW_L + PW_R) / SW + 1;  // output width
+        SW = std::stoi(str_strides[1]);         // weight-wise stride
+    
+    if (node.GetAttr<std::vector<std::string>>("data_layout")[0].substr(0,4)=="NCHW"){
```

Comment:
Is there an assumption that the input should be either "NHWC“ or "NCHW" here?

Reply:
Assume the default layout to be "NHWC".

Reply:
Is there any possible that a blocked layout was passed to this function?

Reply:
According to the current experiments, the possible blocked layout contains: "NCHW*c", "OIHW", "OIHW*i*o" (for onednn v1.7 on AVX-512 and both version on AVX-2), "NHWC", "OHWI*o" (for onednn v2.4 on AVX-512). These are all supported.